### PR TITLE
feat(engineer): Add support for webpack hot reloading

### PR DESCRIPTION
**Pull request by AdamYahid**
Adds hot reloading for react (only) modules when in development mode, using react-refresh

References
https://github.com/pmmmwh/react-refresh-webpack-plugin/tree/6379b3f6ffcb1be02ea6e6012c8332844d3335df/examples/webpack-hot-middleware

### CODE REVIEW

**Comment by AviVahl** [comment]
should be by flag. nothing in the engine forces apps to use React, or even to be a client app for that matter.

**Comment by AdamYahid** [comment]
Yeah there are some API opens here still, but it will be a flag.

File: packages/scripts/src/create-webpack-configs.ts [hunk 1]
```diff
@@ -179,6 +184,20 @@ export function createWebpackConfig({
                 ]
             );
         }
+        if (webpackHot) {
+            // we need the hot middleware client into the env entry
+            entry[envName] = ['webpack-hot-middleware/client', entryPath];
+            plugins.push(new webpack.HotModuleReplacementPlugin());
+            plugins.push(
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+                new ReactRefreshWebpackPlugin({
```
Comment (AdamYahid) [inline packages/scripts/src/create-webpack-configs.ts:193]: @AviVahl @RomanYarik  I'm not really sure about this plugin here
everything I did up to this point is related to webpack hot reloading in general, and is only applicable to applications which have the flag turned on and a web env, so I feel fine about that - along with the fact that we actually are opinionated about webpack

This plugin is only applicable for react applications and should run in conjuncture with the babel/typescript transformation in the users development webpack config, and it feels a bit too opinionated on out part to set it up for them

I would consider documenting something in the lines of
To set up webpack hot reloading for your project you should
1. add ReactRefreshWebpackPlugin to your webpack config
2. Add the babel/typescript transformation to your webpack config
3. engineer start --webpackHot true

What do you think?

Comment (AdamYahid) [reply packages/scripts/src/create-webpack-configs.ts]: Actually while adding this comment I decided to move on with it, in the next commit I moved the react part of hot reloading to the responsibility of the user - that way the can decide the hot reloading solution, we just add the webpack infra for them

File: examples/react/webpack.config.hot.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const { StylableWebpackPlugin } = require('@stylable/webpack-plugin');
```
Comment (RomanYarik) [inline examples/react/webpack.config.hot.js:1]: inherit from base one and just extend

File: packages/scripts/src/create-webpack-configs.ts [hunk 3]
```diff
@@ -117,6 +117,8 @@ interface ICreateWebpackConfigOptions {
     overrideConfig?: TopLevelConfig | TopLevelConfigProvider;
     externalFeaturesRoute: string;
     eagerEntrypoint?: boolean;
+    watch?: boolean;
```
Comment (RomanYarik) [inline packages/scripts/src/create-webpack-configs.ts:120]: legacy code?

**Review by RomanYarik — APPROVED** [review]

File: packages/scripts/src/create-webpack-configs.ts (hunk 3 above)
Comment (AdamYahid) [reply packages/scripts/src/create-webpack-configs.ts:120]: removed

**Pull request merged**